java/flink/src/main/java/com/logicalclocks/hsfs/flink/engine/PojoSerializer.java

```diff
@@ -0,0 +1,59 @@
+/*
+ *  Copyright (c) 2023. Hopsworks AB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ *  See the License for the specific language governing permissions and limitations under the License.
+ *
+ */
+
+package com.logicalclocks.hsfs.flink.engine;
+
+import lombok.SneakyThrows;
+import org.apache.avro.Schema;
+import org.apache.avro.io.BinaryEncoder;
+import org.apache.avro.io.EncoderFactory;
+import org.apache.avro.reflect.ReflectData;
+import org.apache.avro.reflect.ReflectDatumWriter;
+import org.apache.flink.api.common.serialization.SerializationSchema;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class PojoSerializer implements SerializationSchema<Object> {
+
+  @SneakyThrows
+  @Override
+  public byte[] serialize(Object input) {
+    return encode(input);
+  }
+
+  private byte[] encode(Object input) throws IOException {
+
+    Schema schema = ReflectData.get().getSchema(input.getClass());
```

Comment:
We discussed on Slack. This should be using the feature group avro schema, so that messages can be validated before being sent into Kafka.

Additionally this code does not account for complex features (i.e. arrays) which we have to serialize twice.

Reply:
done